modules/core/test/unit/externalApi.ts

```diff
@@ -0,0 +1,112 @@
+const Wallet = require('../../src/wallet');
+import { TestBitGo } from '../lib/test_bitgo';
+import * as _ from 'lodash';
+import * as Bluebird from 'bluebird';
+const co = Bluebird.coroutine;
+import * as nock from 'nock';
+import * as common from '../../src/common';
+
+nock.disableNetConnect();
+const externalHost = 'https://externalsign.com';
+const externalApiPath = '/api/sign';
+
+
+describe('Wallet External API:', function () {
+
+  const userKeypair = {
+    xprv: 'xprv9s21ZrQH143K2fJ91S4BRsupcYrE6mmY96fcX5HkhoTrrwmwjd16Cn87cWinJjByrfpojjx7ezsJLx7TAKLT8m8hM5Kax9YcoxnBeJZ3t2k',
+    xpub: 'xpub661MyMwAqRbcF9Nc7TbBo1rZAagiWEVPWKbDKThNG8zqjk76HAKLkaSbTn6dK2dQPfuD7xjicxCZVWvj67fP5nQ9W7QURmoMVAX8m6jZsGp',
+    rawPub: '02c103ac74481874b5ef0f385d12725e4f14aedc9e00bc814ce96f47f62ce7adf2',
+    rawPrv: '936c5af3f8af81f75cdad1b08f29e7d9c01e598e2db2d7be18b9e5a8646e87c6',
+    path: 'm',
+    walletSubPath: '/0/0',
+  };
+  const backupKeypair = {
+    xprv: 'xprv9s21ZrQH143K47sEkLkykgYmq1xF5ZWrPYhUZcmBpPFMQojvGUmEcr5jFXYGfr8CpFdpTvhQ7L9NN2rLtsBFjSix3BAjwJcBj6U3D5hxTPc',
+    xpub: 'xpub661MyMwAqRbcGbwhrNHz7pVWP3njV2Ehkmd5N1AoNinLHc54p25VAeQD6q2oTS3uuDMDnfnXnthbS9ufC8JVYpNnWU5Rn3pYaNuLCNywkw1',
+    rawPub: '03bbcb73997977068d9e36666bbd5cd37579acae8e2bd5ce9d0a6e5c150a423bc3',
+    rawPrv: '77a15f14796f4001d1092ae84f766bd869e9bee6bffae6547def5045b96fa943',
+    path: 'm',
+    walletSubPath: '/0/0',
+  };
+  const bitgoKey = {
+    xpub: 'xpub661MyMwAqRbcGQcVFiwcrtc7c3vopsX96jsJUYPcFMREcRTqAqsqbv2ZRyCJAPLm5NMHCy85E3ZwpT4EAUw9WGU7vMhG6z83hDeKXBWn6Lf',
+    path: 'm',
+    walletSubPath: '/0/0',
+  };
+
+  describe('Transaction Signature Verification', function () {
+    it('signs p2sh using external api', co(function *() {
+      const prodBitgo = new TestBitGo({ env: 'prod', externalSignerUrl: `${externalHost}${externalApiPath}` });
+      prodBitgo.initializeTestVars();
+      const bgUrl = common.Environments[prodBitgo.getEnv()].uri;
+      const fakeProdWallet = new Wallet(prodBitgo, { id: '2NCoSfHH6Ls4CdTS5QahgC9k7x9RfXeSwY4', private: { keychains: [userKeypair, backupKeypair, bitgoKey] } });
+
+      const firstSignature = '010000000144dea5cb05425f94976e887ccba5686a9a12a3f49710b021508d3d9cd8de16b801000000b600473044022021fa73d5fe61ac8942cd70ff4507c574677ce747de5bc46c3dd2e38ec2448fce022047906d2c0154337ab96041e8fb58c243b9bce5f8818fa991643c1260a1859ad80100004c695221031cd227e40ad61b4e137109cb2845eb6f5a584ed5c67d9d3135cdaa5045a842ea2103a2e7b54c7b2da0992555353b8e26c6acff4248f4351f08787bf3e2efc94b658321025c2a6cde33c2d73ccf12eecf64c54f08f722c2f073824498950695e9883b141253aeffffffff02e803000000000000116a0f426974476f2070327368207465737422a107000000000017a914d039cb3344294a5a384a5508a006444c420cbc118700000000';
```

Comment:
Please move hard coded data to test fixture. Tests should contain only the test logic, and the test data is just noise that isn't relevant to that.

Reply:
Also, these are transactions and not signatures, so this naming is misleading. Also how come we have two separate transactions but only one test?